### PR TITLE
get artist and title from title if no tag

### PR DESCRIPTION
**Pull request by lakedai**
supported filename format: %n.%p-%title, %n.%title, %p-%t
remove "Unknown Artist" artist

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+edit src/ol_metadata.c to support title parse
```
Comment (PedroHLC) [inline readme.md:1]: We already have a `README.md`, no need for a second one. Please, merge them...

File: src/ol_metadata.c [hunk 2]
```diff
@@ -138,6 +138,54 @@ ol_metadata_new_from_variant (GVariant *variant)
     }
   }
   g_variant_iter_free (iter);
+
+  // add lakedai 2020/10/02，parse title to help search
```
Comment (PedroHLC) [inline src/ol_metadata.c:142]: 1 - remove your name from this comment, git will remember who did this changes, and you'll be listed as a contributor in the repository;
2 - add to the comment, what are the conditions for this to trigger.

File: src/ol_metadata.c [hunk 3]
```diff
@@ -138,6 +138,54 @@ ol_metadata_new_from_variant (GVariant *variant)
...
+    }
+    const char * oldartist = ol_metadata_get_artist (metadata);
+    if (g_str_equal (oldartist, "未知艺术家") 
+      || g_str_equal (oldartist, "Unknown Artist")
+      || (g_str_equal (oldartist, "群星") && ! g_str_equal (artist, "")))
+    {
+      ol_metadata_set_artist (metadata, artist);
+    }
+  }
+  // add end
```
Comment (PedroHLC) [inline src/ol_metadata.c:187]: Remove this last comment

**Review by PedroHLC — CHANGES_REQUESTED** [review]
I have a really big problem with the `%p-%t` format: "Title - Live Somewhere" songs...

Could you do it, triggering only when there is no Author (or when it's "Unknown") instead?

**Comment by lakedai** [comment]
> I have a really big problem with the `%p-%t` format: "Title - Live Somewhere" songs...
> 
> Could you do it, triggering only when there is no Author (or when it's "Unknown") instead?



> I have a really big problem with the `%p-%t` format: "Title - Live Somewhere" songs...
> 
> Could you do it, triggering only when there is no Author (or when it's "Unknown") instead?

ok. your problem is solved in the new commit.
this function is only triggered on search action, both auto and manual.
when tag artist is null, empty, or contains any of the following words, it is considered invalid.
-  "unknown" (case-insensitive) 
-  "未知"
-  “群星"
if tag artist is invalid, artist and title are got from tag title.
whether tag artist is valid or not, track number is always removed from title
when tag title is parsed, the following formats are supported: 
-  %n.%p-%t
-  %n.%t--%p
-  %n.%t
-  %p-%t
-  %t--%p
-  %t

**Review by PedroHLC — APPROVED** [review]
Ok, the changes are nice.
I'll test it and then merge it :smile:

**Comment by PedroHLC** [comment]
:sweat_smile:  I should have squashed it before :joy:

**Pull request merged**